### ISSUE TITLE
Cannot specify max threads allowed within TransferThreadPool

### ISSUE THREAD

**Issue by JoeMaher** [aws/aws-sdk-android#241]
* What service are you using?

s3

* In what version of SDK are you facing the problem?

2.3.8

* Is the issue limited to Simulators / Actual Devices?

Both

* Can your problem be resolved if you bump to a higher version of SDK?

No

* Is this problem related to specific Android/OS version?

No

* Can you give us steps to reproduce with a minimal, complete, and verifiable example? Please include any specific network conditions that might be required to reproduce the problem.

Upon `TransferUtility` initialisation it would be preferable to be able to specify the max amount of threads to be used within the `ThreadPoolExecutor` rather than hardcoding it to the lesser of `10` and `Runtime.getRuntime().availableProcessors() + 1` as more than one concurrent upload on a flaky internet connection, especially with largeish files, can become problematic

**Comment by minbi** [comment]
Hi JoeMaher,

Thank you for your feedback. We will take this into consideration for future releases.

**Comment by kvasukib** [comment]
Hi @JoeMaher, We have added an option to specify the number of threads in the thread pool created by TransferUtility. You can take a look at the documentation here: https://docs.aws.amazon.com/aws-mobile/latest/developerguide/how-to-transfer-files-with-transfer-utility.html#native-transfer-utility-options

Example:

```java
TransferUtilityOptions options = new TransferUtilityOptions();
options.setTransferThreadPoolSize(8);

TransferUtility transferUtility = TransferUtility.builder()
    // Pass-in S3Client, Context, AWSConfiguration/DefaultBucket Name
    .transferUtilityOptions(options)
    .build();
```

Sorry for the delayed response.

**Comment by minbi** [comment]
This issue is resolved by version 2.6.16
